src/project/grouped_environment.rs

```diff
@@ -137,13 +140,36 @@ impl<'p> GroupedEnvironment<'p> {
         }
     }
 
+    pub fn platforms(&self) -> HashSet<Platform> {
+        match self {
+            GroupedEnvironment::Group(group) => group
+                .environments()
+                .flat_map(|env| env.platforms())
+                .collect(),
+            GroupedEnvironment::Environment(env) => env.platforms(),
+        }
+    }
+
     /// Returns true if the group has any Pypi dependencies.
     pub fn has_pypi_dependencies(&self) -> bool {
         match self {
             GroupedEnvironment::Group(group) => group.has_pypi_dependencies(),
             GroupedEnvironment::Environment(env) => env.has_pypi_dependencies(),
         }
     }
+
+    /// Returns the features of the group
+    pub fn features(
+        &self,
+    ) -> Either<
+        impl Iterator<Item = &'p Feature> + DoubleEndedIterator + 'p,
+        impl Iterator<Item = &'p Feature> + DoubleEndedIterator + 'p,
+    > {
```

Comment:
```suggestion
    ) ->impl Iterator<Item = &'p Feature> + DoubleEndedIterator + 'p {
```

Reply:
Either already implements Iterator if both its fields implement Iterator. That way you can simplify the type.

Reply:
Oh cool, was already confused by this way of writing it.